File: tests/fixtures/supabase/supabase-realtime-missing-filter-fixed/01-filtered-subscription.ts
import { supabase } from '../client.js';

export function subscribeInbox(userId: string, fetchMessages: () => void) {
  return supabase
    .channel('messages-inbox')
    .on(
      'postgres_changes',
      { event: '*', schema: 'public', table: 'messages', filter: `receiver_id=eq.${userId}` },
      () => fetchMessages(),
    )
    .subscribe();
}

File: tests/fixtures/supabase/supabase-realtime-missing-filter-fixed/01_filtered_subscription.py
from lib.client import supabase


def handle_record_updated(payload):
    print("update", payload)


def subscribe_messages(user_id: str):
    return (
        supabase.channel("room1")
        .on_postgres_changes(
            "UPDATE",
            schema="public",
            table="messages",
            filter=f"receiver_id=eq.{user_id}",
            callback=handle_record_updated,
        )
        .subscribe()
    )
